modules/library/main/src/main/java/org/geotools/styling/DefaultResourceLocator.java

```diff
@@ -84,9 +82,14 @@ protected URL validateRelativeURL(URL relativeUrl) {
         return null;
     }
 
-    URL makeRelativeURL(String uri) {
+    URL makeRelativeURL(String uri, String query) {
```

Comment:
Mind, the GeoServer version uses the other makeRelative method (which is protected). As far as know, that path will strip away the query string (even if by getting  a URL, it has all the needed information, believe the implementation needs to be changed to preserve the query string)